### PR TITLE
Correct output handling for HyPhy-BGM.

### PR DESCRIPTION
**Pull request by davebx**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR updates an existing tool or tool collection

### CODE REVIEW

File: tools/hyphy/hyphy_bgm.xml [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-<tool id="hyphy_bgm" name="HyPhy-BGM" version="@VERSION@+galaxy1" profile="19.09">
+<tool id="hyphy_bgm" name="HyPhy-BGM" version="@VERSION@+galaxy2" profile="19.09">
     <description>- Detecting coevolving sites via Bayesian graphical models</description>
     <macros>
         <import>macros.xml</import>
     </macros>
     <expand macro="requirements"/>
     <command detect_errors="exit_code"><![CDATA[
         @SYMLINK_FILES@
+        ln -s '$bgm_output' ${input_file}.BGM.json &&
```
Comment (bgruening) [inline tools/hyphy/hyphy_bgm.xml:10]: this is not good @davebx ... this is creating a link in the Galaxy DB directory isn't it? So you leave the CWD and Galaxy will never cleanup those dangling symblinks.

Comment (davebx) [reply tools/hyphy/hyphy_bgm.xml:10]: Oh no, other way around, it's creating a symlink in the working directory that points to the galaxy dataset.

Comment (bgruening) [reply tools/hyphy/hyphy_bgm.xml:10]: `${input_file}` is of type /data/db/56/6/6765765876asdf75765a7sdf.dat and you append `.BGM.json` isn't it?

Comment (davebx) [reply tools/hyphy/hyphy_bgm.xml:10]: From `man ln`:

`ln [OPTION]... [-T] TARGET LINK_NAME` where in this case TARGET is ${bgm_output} and LINK_NAME is ${input_file}.BGM.json

The `@SYMLINK_FILES@` token sets `$input_file` to either `input.fa`, `input.fa.gz`, or `input.nexus` depending.

Comment (bgruening) [reply tools/hyphy/hyphy_bgm.xml:10]: Oh I see, you are overwriting "input_file". `input_file` is also a parameter name, that you overwrite with `set`. Ok that was confusing me.

**Review by bgruening — APPROVED** [review]

**Pull request merged**